test/test_kubeclient.rb

```diff
@@ -483,6 +616,26 @@ def test_api_basic_auth_failure
                      times: 1)
   end
 
+  def test_api_basic_auth_failure_raw
+    error_message = 'HTTP status code 401, 401 Unauthorized'
+    response = OpenStruct.new(code: 401, message: '401 Unauthorized')
+
+    stub_request(:get, 'http://username:password@localhost:8080/api/v1')
+      .to_raise(KubeException.new(401, error_message, response))
+
+    client = Kubeclient::Client.new(
+      'http://localhost:8080/api/',
+      auth_options: { username: 'username', password: 'password' }
+    )
+
+    exception = assert_raises(KubeException) { client.get_pods(as: :raw) }
+    assert_equal(401, exception.error_code)
+    assert_equal(error_message, exception.message)
+    assert_equal(response, exception.response)
+
+    assert_requested(:get, 'http://localhost:8080/api/v1', times: 1)
```

Comment:
This test needs to check against the new URL you changed it to in line `623` (will fix tests):

```ruby
assert_requested(:get, 'http://username:password@localhost:8080/api/v1', times: 1)
```